src/execution/sql/ddl_executors.cpp

```diff
@@ -150,10 +152,20 @@ bool DDLExecutors::CreateIndex(const common::ManagedPointer<catalog::CatalogAcce
   const auto &schema = accessor->GetIndexSchema(index_oid);
   // Instantiate an Index and update the pointer in the Catalog
   storage::index::IndexBuilder index_builder;
+  index_builder.SetSqlTableAndTransactionContext(accessor->GetTable(table), populate_txn);
   index_builder.SetKeySchema(schema);
   auto *const index = index_builder.Build();
+  index->SetNotLive();
+
   bool result UNUSED_ATTRIBUTE = accessor->SetIndexPointer(index_oid, index);
   TERRIER_ASSERT(result, "CreateIndex succeeded, SetIndexPointer must also succeed.");
+
+  // Now, populate the index
+  index_builder.BulkInsert(index);
```

Comment:
is it better to bulkinsert everything in one giant transaction (if index is large), or divide it up into multiple smaller transactions?